renku/core/models/projects.py

```diff
@@ -163,6 +167,10 @@ def from_jsonld(cls, data, client=None):
 
     def to_yaml(self):
         """Write an instance to the referenced YAML file."""
+        from renku import __version__
+
+        self.agent_version = __version__
```

Comment:
Should we do this in a @pre_dump marshmallow hook on `ProjectSchema` ? I'm not entirely sure if that's a good idea, on the one hand it'd always set it even if project is nested somewhere, but there might be some circumstances where we don't want that. But I an't think of any off the top of my head.

Reply:
I personally do like fat-serializers & thinner models, however, if you look at our models currently and the current state we have thin-serializers and very fat models were we do all (r/w) model mutations on the model itself. If we want to change that, then we should create a refactoring epic and go for the switch. Not sure if mixing them is the best thing here, it just leads to confusion IMHO (having read mutation on models and then write on serializers). 

I would leave it like this for now and open a ticket to make serializers as fat as possible.